### ISSUE TITLE
Add dependencies from root as peer-dependencies in extension

### ISSUE THREAD

**Issue by sgraband** [eclipse-cdt-cloud/theia-trace-extension#1017]
While updating the [CDT Cloud blueprint](https://github.com/eclipse-cdt-cloud/cdt-cloud-blueprint) to the latest version of this extension i noticed that some dependencies do not seem to be provided correctly.

To be more specific: It seems like these [dependencies from the root package.json](https://github.com/eclipse-cdt-cloud/theia-trace-extension/blob/a1d5a0cdb5ca00db86107e0f7847a76e20e70786/package.json#L128-L130) are only references as dependencies there, but not in the `theia-traceviewer` extensions `package.json`.

Therefore, i need to manually add these dependencies to my project, when i want to consume the `theia-traceviewer` extension.

These dependencies should probably be added as peer dependencies in the `theia-traceviewer package.json`. Or was there a specifiic reason for it to be implemented this way?

**Comment by bhufmann** [comment]
> While updating the [CDT Cloud blueprint](https://github.com/eclipse-cdt-cloud/cdt-cloud-blueprint) to the latest version of this extension i noticed that some dependencies do not seem to be provided correctly.
> 
> To be more specific: It seems like these [dependencies from the root package.json](https://github.com/eclipse-cdt-cloud/theia-trace-extension/blob/a1d5a0cdb5ca00db86107e0f7847a76e20e70786/package.json#L128-L130) are only references as dependencies there, but not in the `theia-traceviewer` extensions `package.json`.
> 
> Therefore, i need to manually add these dependencies to my project, when i want to consume the `theia-traceviewer` extension.
> 
> These dependencies should probably be added as peer dependencies in the `theia-traceviewer package.json`. Or was there a specifiic reason for it to be implemented this way?

No, I don't think there was a specific reason for that. When the change was introduced I think it was not noticed or not questioned. If you think that's incorrect, please provide a PR and we will review it. 

Thanks for bringing this up.

**Comment by marcdumais-work** [comment]
I believe that this is a legit issue - the way these production dependencies are defined ATM, they will be available to the test application but not be included in the published Theia extension. If they are only needed in the extension, I think they should be moved there.
